### PR TITLE
User login system

### LINKED ISSUE #33: create user account
Create accounts structure in database and a function to create new accounts. 

A user account consists of:
- a unique id
- a unique email
- password (hashed & salted)
- a usergroup

### PR DESCRIPTION
**Pull request by SpaceLenore**
This adds a complete user login system and accounts with both users and admins. Admins can create new accounts. Closes #33 #36 #37 #38 #39 #40 #41 #44

### CODE REVIEW

File: router/login.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    return res.json({
+        "info": "If the user exists, a temporary token has been created." +
+        "You should recive an email shortly.",
+        "error": false
+    });
+});
+
+//Sets a new password with a one time key confirmation
+//Takes a POST request with  x-www-form-urlencoded data and sets a new password
+router.post("/passwordreset", urlencodedParser, async (req, res) => {
```
Comment (Glassinatorn) [inline router/login.js]: If we're supposed to be picky, we probably should not have the reset url be predictable. 
Probably not necessary in our case though. But still.

File: src/loginHandler.js [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+    //New password must be written twice and match
+    if (userChangePassword[0].newPassword != userChangePassword[0].confirmNewPassword) {
+        return false;
+    }
+
+    //Create new hashed password and salt and update the database
+    let newpw = await createNewPasswordHash(userChangePassword[0].newPassword);
+    let res = await dbo.collection('Users').updateOne(
+        {"username": userChangePassword[0].username},
+        {$set: {"password": newpw}}
```
Comment (Glassinatorn) [inline src/loginHandler.js]: First terrified but then saw passwd hash

File: src/smtpMailman.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+* and return error code 1.
+*/
+const testSmtpConnection = () => {
+    let transporter = nodemailer.createTransport({
+        host: emailConfig.smtp.host,
+        port: emailConfig.smtp.port,
+        secure: false, // upgrade later with STARTTLS
+        requireTLS: true,
+        tls: {
+            ciphers: 'SSLv3'
```
Comment (Glassinatorn) [inline src/smtpMailman.js]: *thumbs up*

File: router/login.js (hunk 1 above)
Comment (SpaceLenore) [reply router/login.js]: I understand your intentions and thought process but in this case it doesn't matter. The route takes a unique one time token, username, new password, and confirmation of new password. Token must be valid for the specified user and both new passwords must match. The route is not unique for each user but the data provided in the `POST` request is.

File: API_DOC.md [hunk 4]
```diff
@@ -19,3 +35,41 @@ Fetch items of a specific category based on the `Namn` field in the database wit
...
+### Password reset
+With the one time reset token from the requestreset route the user can set a new password.
+Send a `POST` request to `/acc/passwordreset` with the form data for `username`, `oneTimeKey`,
+`newPassword` and `confirmNewPassword`
+
+## Admin
+This handles all the admin routes and cannot be used by non-admins. All routes are
+available over at `/admin/`
+
+### user
```
[the remark this answers is not in the document]
Comment (SpaceLenore) [reply API_DOC.md]: fixed in comit 3e37e21

**Pull request merged**